proxy/src/set_public_ip_and_start.sh

```diff
@@ -31,9 +31,28 @@ then
     fi
 fi
 
+## PUBLIC_IP retrieved from third-party sources
+if [[ $PUBLIC_IP == '' ]]
+then
+    urls=(
+        'https://icanhazip.com/'
+        'https://ipinfo.io/ip'
+        'https://domains.google.com/checkip'
+    )
+
+    # Attempt retrieval of the public ip from the third-party sources
+    for url in "${urls[@]}"; do
+        PUBLIC_IP="$(curl --max-time 2 -s "${url}")"
```

Comment:
Maybe you should check here to see if the `PUBLIC_IP` obtained then break the loop. There is no point to redo the same action in loop and specially if the last one is failed you would override the null with the IP you found in the first iterations.

Reply:
@nghazali Agree, just updated the PR, thanks.